app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,89 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if logged_in?
+      @tweets = Tweet.all
+      erb :'tweets/tweets'
+    else
+      flash[:notice] = 'Please log in first to view tweets.'
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      # @user = current_user
+      erb :'/tweets/create_tweet'
+    else
+      flash[:notice] = 'Please log in first to view tweets.'
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'/tweets/show_tweet'
+    else
+      flash[:notice] = 'Please log in first to view tweets.'
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == current_user.id
+        erb :'/tweets/edit_tweet'
+      else
+        flash[:notice] = 'You can only edit your own tweets.'
+        redirect to "/tweets/#{params[:id]}"
+      end
+    else
+      flash[:notice] = 'Please log in first to view tweets.'
+      redirect to '/login'
+    end
+  end
+
+  patch '/tweets/:id' do
+    if logged_in?
+      #compare tweet.user_id to user.id
+      tweet = current_user.tweets.find_by(id: params[:id])
+      # tweet = Tweet.find_by_id(params[:id])
+      if params.value?('')
+        flash[:notice] = 'Cannot create empty tweet.'
+        redirect to "/tweets/#{params[:id]}/edit"
+      else
+        tweet.update(content: params[:content])
+        redirect to '/tweets'
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params.value?('')
+      flash[:notice] = 'Cannot create empty tweet.'
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.